### PR TITLE
Fix 'Tried x times' message

### PR DESCRIPTION
**Pull request by webrouse**
Jira: https://keboola.atlassian.net/browse/COM-681

### CODE REVIEW

File: src/BaseExportAdapter.php [hunk 1]
```diff
@@ -76,11 +77,12 @@ protected function handleDbError(Throwable $e, int $maxRetries, ?string $outputT
         $message = $outputTable ? sprintf('[%s]: ', $outputTable) : '';
         $message .= sprintf('DB query failed: %s', $e->getMessage());
 
-        // Retry mechanism can be disabled
-        if ($maxRetries > 1) {
+        // Retry mechanism can be disabled or modified, ... we need to print real count
+        if ($e instanceof UserRetriedException && $e->getTryCount() > 1) {
             $message .= sprintf(' Tried %d times.', $maxRetries);
```
Comment (webrouse) [inline src/BaseExportAdapter.php:82]: - I want to disable retry mechanism in ex-hive for `SemanticException`.
- But I need to fix this bad message first.
- This message print the number of retries from the configuration, regardless of whether the retry have actually been performed so many times (it works well for the default `SimpleRetryPolicy` implementation, but not for the custom one).
- Therefore, the number of retries is extracted from the exception to which I added this value.

File: src/Connection/BaseDbConnection.php [hunk 2]
```diff
@@ -120,13 +120,12 @@ protected function connectWithRetry(): void
...
+        $proxy = $this->createRetryProxy($maxRetries);
         try {
-            return $this
-                ->createRetryProxy($maxRetries)
-                ->call($callback);
+            return $proxy->call($callback);
         } catch (Throwable $e) {
             throw in_array(get_class($e), $this->getExpectedExceptionClasses(), true) ?
-                new UserRetriedException($e->getMessage(), 0, $e) :
+                new UserRetriedException($proxy->getTryCount(), $e->getMessage(), 0, $e) :
```
Comment (webrouse) [inline src/Connection/BaseDbConnection.php:128]: - The number of retries is returned from the proxy.

File: src/Exception/DeadConnectionException.php [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 namespace Keboola\DbExtractor\Adapter\Exception;
 
-class DeadConnectionException extends UserRetriedException
+class DeadConnectionException extends UserException
```
Comment (webrouse) [inline src/Exception/DeadConnectionException.php:7]: DeadConnection is retried ... but it is low level exception wrapped to `UserRetriedException`, so I had to modify the type.

File: tests/phpunit/AbstractExportAdapterTest.php [hunk 4]
```diff
@@ -87,9 +88,10 @@ public function testExportFailed(): void
         try {
             $exportAdapter->export($config, $this->getCsvFilePath());
             $this->fail('Exception expected');
-        } catch (UserExceptionInterface $e) {
+        } catch (UserRetriedException $e) {
             Assert::assertStringContainsString('[output]: DB query failed:', $e->getMessage());
             Assert::assertStringContainsString("Tried $retries times.", $e->getMessage());
+            Assert::assertSame($e->getTryCount(), $retries);
```
Comment (webrouse) [inline tests/phpunit/AbstractExportAdapterTest.php:94]: Test.

**Comment by webrouse** [comment]
Hi @pivnicek , can you please review It? It is related to ZD ticket, see comments.

**Review by pivnicek — APPROVED** [review]
I think it should be ok.

good idea to carry the try count in the exception 👍

**Pull request merged**